docker-compose-prod.yml

```diff
@@ -0,0 +1,56 @@
+version: '3.0'
+
+services:
+  sdschedule-database:
+    container_name: sdschedule-database
+    image: mariadb:latest
+    environment:
+      - "MYSQL_ROOT_PASSWORD=password" 
+      - "MYSQL_DATABASE=classes"
+    volumes:
+      - "sdschedule-data:/var/lib/mysql"
+  sdschedule-backend:
+    container_name: sdschedule-backend
+    build: backend 
+    ports:
+      - "5000:5000"
+    depends_on: 
+      - "sdschedule-database"
+    environment:
+      - "PYTHONUNBUFFERED=0"
+    volumes:
+      - "./backend:/app"
+  sdschedule-frontend:
+    container_name: sdschedule-frontend
+    build: frontend
+    depends_on: 
+      - "sdschedule-backend"
+    volumes:
+      - "./frontend/src:/app/src"
+      - "./frontend/public:/app/public"
+      - "sdschedule-frontend-build:/app/build"
+    environment:
+      - "ENV=PROD"
+  web:
+    container_name: sdschedule-web
+    image: nginx
+    ports:
+      - "80:80"
+      - "443:443"
+      # - "3000:3000"
+    depends_on:
+      - "sdschedule-frontend"
+    volumes:
+      - "sdschedule-frontend-build:/var/www/sdschedule"
+      #- "./web/key:/etc/nginx/ssl-key:ro" # for cloudflare origin ssl
+      #- /var/run/docker.sock:/tmp/docker.sock:ro
+      - "./web/letsencrypt:/etc/letsencrypt:ro"
```

Comment:
Could you explain what all these files are? I think they are trying to copy over the nginx configs from the host machine to the container

Reply:
line 44: Our front-end container will build the compressed production codes and save them to a docker volume shared with nginx container. Front-end container will quit after build.

line 45: If we use SSL from cloudflare, this is the SSL used between our server and cloudflare.

line 46: one of the configs required to use let's encrypt SSL

line 47: let's encrypt

----------

Generally they are not "copying", but "sharing". Since everything will be in container, including letsencrypt, different containers need a shared directory.